### DISCUSSION TITLE
GitLab vs GitHub

### DISCUSSION THREAD

**Discussion by mattjbr123** [ukceh-rse/community#10] [Social Coding]
### I'd like to work on...

Not completely sure this is the right place to post this but I figured it can always be moved if not! I'm kinda using this discussion forum as a Q&A xD

Today I'd like to know the difference between and mertis of using GitLab vs GitHub. The context being I'm familiar with and use GitHub everyday but see a few folks using GitLab, and have so far been a bit "Ew it's different why do I have to use it for X project", a reaction I can see many people having. 
From what I've gleaned here and there it's very similar in functionality to GitHub but is self-hosted/not hosted by Microsoft. I'm sure there must be more to it than this, and would love to know! 

### Collaboration

I would like to work as a group.

### My aim is to...

Not be put off by the fact it looks a bit different to GitHub, basically xD

And to be able to make informed decisions about which I should be using for what!

And to know enough to maybe summarise the results of this discussion in a blog post to help people who feel similarly to me

**Comment by metazool** [comment]
I remember talking to @rodscott about this awhile ago - we have a self-hosted Gitlab (as a lot of related / partner organisations do) for more infrastructure-oriented projects.

The message I came away with is the self-hosted Gitlab instance is _deprecated_ though I don't know if this is an official status or just a feeling about it :)  ... one can see how this would involve

1. being conscious and communicative about Gitlab deprecation
2. guidance for how and where to start new projects somewhere that isn't Gitlab
3. migration plan for infrastructure projects that depend on it (EIDC is the big one)

On a personal level... I still have an account on the (cloud-based) gitlab.org which dates back to the last time I was working in an organisation that had the same mix of private Gitlab, public Github. I tried keeping a few personal projecrs there in a spirit of "don't want all the infrastructure to be owned by Microsoft" but over time I've drifted back to keeping new projects in Github. This is partly down to Gitlab.com's wholehearted embrace of AI marketing rhetoric - you want to point people at "come here to collaborate with your community", not "AI-powered software development lifecycle" ;) - and partly entropy, everything else is still in Github. Accept it for now, try to keep as light a footprint as possible with the proprietary / closed parts of it (like Actions) and revisit the decision over time (ideally as more UK / EU based, FOSS oriented and distributed infrastructure becomes available - it would be good to hear about what forward-looking organisations in high-investing places like Germany are working on in this area!)

The moral of this ramble is that _not using Github_ has to be a very conscious choice!
